app_feup/lib/redux/action_creators.dart

```diff
@@ -243,35 +244,58 @@ ThunkAction<AppState> getUserSchedule(Completer<Null> action) {
   return (Store<AppState> store) async {
     try {
       store.dispatch(SetScheduleStatusAction(RequestStatus.busy));
-      var date = DateTime.now();
-      final String beginWeek = date.year.toString().padLeft(4, '0') +
-          date.month.toString().padLeft(2, '0') +
-          date.day.toString().padLeft(2, '0');
-      date = date.add(Duration(days: 6));
-
-      final String endWeek = date.year.toString().padLeft(4, '0') +
-          date.month.toString().padLeft(2, '0') +
-          date.day.toString().padLeft(2, '0');
-
-      final List<Lecture> lectures = await parseSchedule(
-          await NetworkRouter.getWithCookies(
+
+      //TODO: Go back to API whenever it is fixed: https://github.com/NIAEFEUP/project-schrodinger/issues/300
+      //       var date = DateTime.now();
+      // final String beginWeek = date.year.toString().padLeft(4, '0') +
+      //     date.month.toString().padLeft(2, '0') +
+      //     date.day.toString().padLeft(2, '0');
+      // date = date.add(Duration(days: 6));
+
+      // final String endWeek = date.year.toString().padLeft(4, '0') +
+      //     date.month.toString().padLeft(2, '0') +
+      //     date.day.toString().padLeft(2, '0');
+      // final List<Lecture> lectures = await parseSchedule(
+      //     await NetworkRouter.getWithCookies(
+      //         NetworkRouter.getBaseUrlFromSession(
+      //                 store.state.content['session']) +
+      // ignore: lines_longer_than_80_chars
+      //             '''mob_hor_geral.estudante?pv_codigo=${store.state.content['session'].studentNumber}&pv_semana_ini=$beginWeek&pv_semana_fim=$endWeek''',
+      //         {},
+      //         store.state.content['session']));
+
+      //https://sigarra.up.pt/feup/pt/hor_geral.estudantes_view?pv_fest_id=1047393
+      //https://sigarra.up.pt/feup/pt/hor_geral.estudantes_view?pv_fest_id=1047393&pv_ano_lectivo=2020&pv_periodos=1
```

Comment:
If you want to leave this in the codebase, maybe we should create a class in the controller that implements an interface "ScheduleFetcher" or something. Have then a "ScheduleFetcherHTML" and a "ScheduleFetcherAPI". Whenever we want to change the fetch method we just switch the fetcher object construction.

Reply:
Will do.

Reply:
Done!